x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/ExpressionBuilder.java

```diff
@@ -767,6 +777,16 @@ private SqlTypedParamValue param(TerminalNode node) {
         return params.get(token);
     }
 
+    private SqlParser.SqlParameter param(ParserRuleContext ctx) {
+        if (!ctx.getStart().equals(ctx.getStop())) {
+            throw new ParsingException(source(ctx), "Single PARAM literal expected");
```

Comment:
Incorrect style (use `== false` instead of `!`).
I'm not sure what this check tries to prevent...

Reply:
style: any reason why this is not part of `checkStyle` and the `precommit` checks?
I will remove this check, it was more like a sanity check (during unit & integ test) for me to see if there are any possibility where the start and stop token won't be the same. This should never happen.